WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewController.swift

```diff
@@ -118,9 +105,9 @@ struct SetUpTapToPayInformationView: View {
                 Spacer()
 
                 Button(Localization.setUpButton, action: {
-                    setUpButtonAction?()
+                    viewModel.setUpTapped()
```

Comment:
💯 extracting the logic to the view model

Reply:
Yeah it's still a bit clumsy with the way they're inited and have dependencies on one another (mostly because of the way/order they're created) but I don't want to change that underlying pattern. Seems like I'll be able to get the actual work done by the vm though, which should be good enough.